### DISCUSSION TITLE
Is it possible to get all the the pages url or slug at once in docusaurus dynamically?

### DISCUSSION THREAD

**Discussion by placeholdergit** [facebook/docusaurus#6728] [Q&A]
The intention is to create a "go to random page" kind of feature in a docusaurus site.

My initial thought is to select a random page given all the urls or slug by some docusaurus api or function. 

Is there a way to achieve this?

**Comment by Josh-Cena** [comment]
In the [`postBuild`](https://docusaurus.io/docs/next/api/plugin-methods/lifecycle-apis#postBuild) plugin lifecycle, you will receive all the routes, which you can then save as a separate JSON file and import in your React app. Within a React app, you can also import from `'@generated/routes'`.
